python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,576 @@
+    changing any non-word characters to underscores.
+    """
+    return re.sub(r"[^\w]", "_", fieldName)
+
+def quoteIdentifier(identifier):
+    """Return a MySQL compatible version of the given identifier by quoting
+    it with back-ticks (and escaping embedded back-ticks).
+    """
+    return "`" + identifier.replace("`", "``") + "`"
+
+def aliasesFor(name, mappings):
+    """Compute and return the set of possible aliases for the given field name.
+
+    The afw table library processes a given name F by replacing the longest
+    prefix of F that can be found in a schema's alias map with the corresponding
+    alias target. Replacement is repeated until either no such prefix is found,
+    or N replacements have been made (where N is the size of the alias map).
+
+    The mappings argument must be a sorted list of substitutions, where a single
+    substitution is a 2-tuple of strings (source, target).
+    """
+    n = 0
+    aliases, names = set(), set()
+    names.add(name)
+    while n < len(mappings) and len(names) > 0:
+        # Perform one round of reverse alias substitution
+        n += 1
+        newNames = set()
+        for name in names:
+            for i, (source, target) in enumerate(mappings):
+                if name.startswith(target):
+                    alias = source + name[len(target):]
+                    # alias is only valid if source is its longest prefix in mappings.
+                    # A prefix strictly longer than source must occur after it in the
+                    # sorted list of mappings.
+                    valid = True
+                    for s, _ in mappings[i + 1:]:
+                        if not s.startswith(source):
+                            break;
+                        if alias.startswith(s):
+                            valid = False
+                            break
+                    if valid:
+                        aliases.add(alias)
+                        newNames.add(alias)
+        names = newNames
+    return aliases
+
+
+class IngestCatalogConfig(pexConfig.Config):
+    """Configuration for the IngestCatalogTask.
+    """
+
+    allowReplace = pexConfig.Field(
+        "Allow replacement of existing rows with the same unique IDs",
+        bool, default=False
+    )
+
+    maxQueryLen = pexConfig.Field(
+        "Maximum length of a query string. None means use a non-standard, "
+        "database-specific way to get the maximum.",
+        int, optional=True, default=None
+    )
+
+    maxColumnLen = pexConfig.Field(
+        "Maximum length of a database column name or alias. Fields "
+        "that map to longer column names will not be ingested.",
+        int, default=64
+    )
+
+    idColumnName = pexConfig.Field(
+        "Name of the unique ID column",
+        str, optional=True, default="id"
+    )
+
+    remap = pexConfig.DictField(
+        "A mapping from normal to desired SQL column names",
```

Comment:
The help here isn't super helpful!

I understand the intention after reading the task documentation, but can we be a bit more specific about what a "normal column name" is?

Also, if I'm correct this maps from canonicalized field name to an alternative. That requires the user to know what the canonicalization algorithm is. Might it be better to map from the pre-canonicalization field name instead?